### ISSUE TITLE
Multiplayer: Should the game be pausable?

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#663]
Or except for the server client. And then, the remote clients should simply follow.

IMHO, less simple than it seems. ;)
We should discuss what we want here.

**Comment by hwoarangmy** [comment]
Yes, IMHO, it is more a question of what behaviour do we want...

**Comment by akien-mga** [comment]
That's a good question actually. I can see pros and cons for letting the host client manage the pause feature. Basically in the current version, since there is no matchmaking feature or server listing, most people will be playing with acquaintances; in such a configuration I think that the pause feature is a nice addition, because one player might ask the host to pause the game 5 min if need be.

In a configuration where players will compete against stranger, I think the feature will become more of a bother if the host client can arbitrarily decide to pause your game.

**Comment by hwoarangmy** [comment]
In SC2, every client can pause the game (up to 3 times). If the game is paused, any client can resume it.

**Comment by akien-mga** [comment]
> In SC2, every client can pause the game (up to 3 times). If the game is paused, any client can resume it.

That could be a nice way to do it, maybe with a small timer before other clients can unpause the game (e.g. 5 seconds to prevent compulsive un-pausing ;)).
If we do it like that, I'd like the chat to stay on when the game is paused (that's pretty obvious, but it doesn't hurt mentioning it).

**Comment by Bertram25** [comment]
There are two roads, IMHO.
Either we add something rather simple, like disabling pausing in multiplayer or simply letting the server-client pause, or we add something more advanced.

WDYT?

If we go the second way:
What's your favorite way to do it? What would useful for you IRL?

**Comment by hwoarangmy** [comment]
IMHO, being able to pause is required. I would not be for a half solution here. The current behaviour is enough IMO. But if we want to go further, we should go for a definitive solution like allowing 3 pauses per client and allowing any client to resume a game.

**Comment by Bertram25** [comment]
I then ask a vote:
Should we either:
- Let it as is?
  or:
- Add 3 pause per player and permit any player to unpause after a few secs.

@oyvindln @Danimal696 @hwoarangmy @akien-mga ?

**Comment by Bertram25** [comment]
oops, I forgot: pinging also @Wout-van-Dongen @eugeneloza

**Comment by eugeneloza** [comment]
I'm afraid, I have never ever played an online game :D Not a single one :D
So as for theoretic: yes, the best solution seems to me unlimited pauses for everybody but anybody can unpause.
Or (more complex): the player may issue a 'please, can we pause the game for ___ minutes?' which has to be approved by every participant. I.e. the pauser must define how long the pause will be.

**Comment by Bertram25** [comment]
Taking @hwoarangmy latest comment. Let's do that Later:

> IMHO, being able to pause is required. I would not be for a half solution here. The current behaviour is enough IMO. But if we want to go further, we should go for a definitive solution like allowing 3 pauses per client and allowing any client to resume a game.
